contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -233,5 +280,46 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
             counterpartyThetaVault.optionAuctionID(),
             orders
         );
+
+        _updatePPS();
+    }
+
+    /**
+     * @notice Updates the price per share of the current round. The current round
+     * pps will change right after call rollToNextOption as the gnosis auction contract
+     * takes custody of a % of `asset` tokens, and right after we claim the tokens from
+     * the action as we may recieve some of `asset` tokens back alongside the oToken,
+     * depending on the gnosis auction outcome. Finally it will change at the end of the week
+     * if the oTokens are ITM
+     */
+    function _updatePPS() private {
+        (, uint256 newPricePerShare, ) =
+            VaultLifecycle.rollover(totalSupply(), vaultParams, vaultState);
+
+        roundPricePerShare[vaultState.round] = newPricePerShare;
+    }
+
+    /**
+     * @notice Withdraws from the most recent deposit which has not been processed
+     * @param amount is how much to withdraw in total
+     * @return the amount left to withdraw
+     */
+    function _withdrawFromNewDeposit(uint256 amount) private returns (uint256) {
+        Vault.DepositReceipt storage depositReceipt =
+            depositReceipts[msg.sender];
+
+        uint256 receiptAmount = depositReceipt.amount;
+
+        // Immediately get what is in the pending deposits, without need for checking pps
+        if (depositReceipt.round == vaultState.round && receiptAmount > 0) {
```

Comment:
we dont need to do an additional check on whether deposit receipt has been processed right. processed is only set to true in redeem. but the condition in redeem is that the round of the deposit is before the current round. but here we are making sure the deposit round is the same as current round. so it would not be possible for it to be processed.

Reply:
Yea we don't have to check it